validator_client/src/notifier.rs

```diff
@@ -85,6 +85,10 @@ async fn notify<T: SlotClock + 'static, E: EthSpec>(
         let proposing_validators = duties_service.proposer_count(epoch);
         let attesting_validators = duties_service.attester_count(epoch);
         let doppelganger_detecting_validators = duties_service.doppelganger_detecting_count();
+        let doppelganger_service_enabled = duties_service
+            .validator_store
+            .doppelganger_protection_enabled();
+        let reactivation_epoch = epoch.saturating_add(3_u64);
```

Comment:
Ohhh okay. got it. Yes the calculating part for each validator seems bit tricky.

Reply:
Maybe I can get all validators like this
`let indices_map = self.compute_detection_indices_map(get_index);`

and then iterating over it to get the doppelganger_state to calculate remaining epochs.

[like this](https://github.com/sigp/lighthouse/blob/stable/validator_client/src/doppelganger_service.rs#L617-L618)

Reply:
Should like you've got it! I would probably not worry about it in this log though because it happens every slot so might be a bit too much info. 

But if we wanted update [this log](https://github.com/sigp/lighthouse/blob/stable/validator_client/src/doppelganger_service.rs#L646) (which runs once per epoch), we could make it use this type of map and only log once for all validators as opposed to once per validator (as it does now).